slf4j-simple/src/main/java/org/slf4j/impl/SimpleLogger.java

```diff
@@ -209,8 +209,15 @@ public static SimpleLogListenerImpl makeLogListener(Hashtable config, PrintStrea
         return new SimpleLogListenerImpl(stream, stackTracePrinter, timeAdjuster, SHOW_DATE_TIME, DATE_FORMATTER, START_TIME, SHOW_THREAD_NAME, LEVEL_IN_BRACKETS, WARN_LEVEL_STRING);
     }
 
+    public static void updateConfig(Hashtable config) {
+        synchronized(SIMPLE_LOGGER_PROPS) {
+            initConfig(config);
+        }
+        reinitializeOldLoggerLevels();
+    }
+
     private static void initConfig(Hashtable config) {
-        SIMPLE_LOGGER_PROPS = config;
+        overwriteHashtable(config, SIMPLE_LOGGER_PROPS);
```

Comment:
Could you elaborate why is it preferred to rewrite the existing `Hashtable` content instead of reassigning a new instance to it as it used to be? (Reassigning seems a bit more atomic than clearing + putting each element separately, because that wouldn't leave a timeframe where the table isn't fully initialized)?

Reply:
Thing to keep in mind:
- The `SIMPLE_LOGGER_PROPS` instance is shared by multiple instances and likely multiple threads
- The reference to the `SIMPLE_LOGGER_PROPS` instance is also shared
- Minor: The contents of the `Hashtable` could change outside this framework (instance was reused when passed into `initConfig` method)

If the instance is reassigned then the field needs to be made `volatile` and possibly reducing performance, maybe tiny performance hit, but still, can be avoided.

> (Reassigning seems a bit more atomic than clearing + putting each element separately, because that wouldn't leave a timeframe where the table isn't fully initialized)
Since access to `Hashtable` is synchronized internally, the logic in `updateConfig` already uses this feature to make the update atomic.

Overall I felt that this solution is tiny-bit better than reassigning the instance.